### ISSUE TITLE
Admin Page Failed Auth Handler 

### ISSUE THREAD

**Issue by padenalex** [dotnet-architecture/eShopOnWeb#525]
Hey, 

More of a question but I spun up the sample and visited /admin but it pends on 'Authorizing....' if I'm not logged in and says 'Not authorized' if I'm in the user account.

Shouldn't these redirect to a general 404 or something for security? Maybe it's just a setting I've missed. Thanks

**Comment by ardalis** [comment]
You have to have the Public API project running for the /admin Blazor App to work properly. Is it (PublicAPI) running?

**Comment by padenalex** [comment]
Yes, I ran off compose. It does work correctly (As far as I can tell) but I
thought good practice was to not specify the error to narrow down attack
vectors

**Comment by ardalis** [comment]
Providing a 401 response for Not Authorized is pretty standard, in my experience, since the client code can the use this response code to redirect the user to the client's login page (which might be built-in, or require loading a separate auth provider via OAuth, etc.). Response codes are part of your API's interface and are not exposing error conditions - they're expected, normal responses that endpoints might return in response to known and anticipated user requests. Bad Request errors are another example of a response code that is frequently returned to help API clients present useful information to users such as which fields are required or otherwise invalid. These aren't server errors (that would be a 500) but simply variations of unsuccessful (but otherwise running as expected) requests made to the API.

Does that help?

**Comment by padenalex** [comment]
Yes, I think I understand.
Although to note- I’m still thinking you wouldn’t want anyone not logged in
as an admin already to know /admin exists as anything other than 404
redirect home.

Like why expose the existence of the page at all (outside of tutorial)?
